rest_framework_json_api/schemas/openapi.py

```diff
@@ -0,0 +1,878 @@
+            'fields': {
+                'name': 'fields',
+                'in': 'query',
+                'description': '[sparse fieldsets]'
+                               '(https://jsonapi.org/format/#fetching-sparse-fieldsets).\n'
+                               'Use fields[\\<typename\\>]=field1,field2,...,fieldN',
+                'required': False,
+                'style': 'deepObject',
+                'schema': {
+                    'type': 'object',
+                    'properties': {
+                        '<typename>': {  # placeholder for actual type names
+                            'type': 'string'
+                        }
+                    }
+                },
+                'explode': True
+            },
+            'sort': {
+                'name': 'sort',
+                'in': 'query',
+                'description': '[list of fields to sort by]'
+                               '(https://jsonapi.org/format/#fetching-sorting)',
+                'required': False,
+                'style': 'form',
+                'schema': {
+                    'type': 'string'
+                }
+            },
+        },
+    }
+
+    def get_schema(self, request=None, public=False):
+        """
+        Generate a JSONAPI OpenAPI schema.
+        Overrides upstream DRF's get_schema.
+        """
+        # TODO: avoid copying so much of upstream get_schema()
+        schema = super().get_schema(request, public)
+
+        components_schemas = {}
+        security_schemes_schemas = {}
+
+        # Iterate endpoints generating per method path operations.
+        paths = {}
+        _, view_endpoints = self._get_paths_and_endpoints(None if public else request)
+
+        #: `expanded_endpoints` is like view_endpoints with one extra field tacked on:
+        #: - 'action' copy of current view.action (list/fetch) as this gets reset for each request.
+        expanded_endpoints = []
+        for path, method, view in view_endpoints:
+            if isinstance(view, RelationshipView):
+                expanded_endpoints += self._expand_relationships(path, method, view)
+            elif hasattr(view, 'action') and view.action == 'retrieve_related':
+                expanded_endpoints += self._expand_related(path, method, view, view_endpoints)
+            else:
+                expanded_endpoints.append((path, method, view,
+                                           view.action if hasattr(view, 'action') else None))
+
+        for path, method, view, action in expanded_endpoints:
+            if not self.has_view_permissions(path, method, view):
+                continue
+            # kludge to preserve view.action as it changes "globally" for the same ViewSet
+            # whether it is used for a collection, item or related serializer. _expand_related
+            # sets it based on whether the related field is a toMany collection or toOne item.
+            current_action = None
+            if hasattr(view, 'action'):
```

Comment:
It's in 
```
    def get_request_body(self, path, method, action=None):
        """
        A request body is required by jsonapi for POST, PATCH, and DELETE methods.
        This has an added parameter which is not in upstream DRF:

        :param action: None for conventional path; 'rel' or 'rels' for a singular or plural
            relationship of a related path, respectively.
        """
```

The action gets changed e.g. from `List` (for GET) to `rel` or `rels` for singular or plural relationships since the return schema is singular `"data": {...}`  vs. plural `"data": [ ... ]`

Reply:
I think I have found the method which relies on a patched view which is `is_list_view`.

Reply:
I confused myself in the prior response.

To be more clear about what's happening here...  With  kwarg`{'get': 'retrieve_related'}` the path that starts at, e.g. AuthorViewSet, but returns the views for each `<related_field>` so I am using the related view's schema but meanwhile `self.view` is still the "parent" View (Authors in this example).
```
    url(r'^authors/(?P<pk>[^/.]+)/(?P<related_field>\w+)/$',
        AuthorViewSet.as_view({'get': 'retrieve_related'}),
        name='author-related'),
```

```yaml
/authors/{id}/comments/:
    # GET of the AuthorView results in the schema for CommentView.
```
And since the relationship can be toOne or toMany, the action is set to either `fetch` or `list` in to get the singular or plural response schema.